src/main/kotlin/no/skatteetaten/aurora/mokey/service/ManagementEndpoint.kt

```diff
@@ -0,0 +1,100 @@
+package no.skatteetaten.aurora.mokey.service
+
+import com.fasterxml.jackson.core.JsonParseException
+import com.fasterxml.jackson.databind.JsonNode
+import com.fasterxml.jackson.databind.node.MissingNode
+import com.fasterxml.jackson.module.kotlin.jacksonObjectMapper
+import no.skatteetaten.aurora.mokey.extensions.asMap
+import no.skatteetaten.aurora.mokey.service.Endpoint.*
+import org.slf4j.Logger
+import org.slf4j.LoggerFactory
+import org.springframework.stereotype.Service
+import org.springframework.web.client.HttpStatusCodeException
+import org.springframework.web.client.RestClientException
+import org.springframework.web.client.RestTemplate
+
+@Service
+class ManagementEndpointFactory(val restTemplate: RestTemplate) {
+    fun create(managementUrl: String): ManagementEndpoint {
+        return ManagementEndpoint.create(restTemplate, managementUrl)
+    }
+}
+
+class ManagementEndpointException(val endpoint: Endpoint, val errorCode: String, cause: Exception? = null)
+    : RuntimeException("${endpoint}_$errorCode", cause)
+
+enum class Endpoint(val key: String) { HEALTH("health"), INFO("info"), MANAGEMENT("_links") }
+
+data class ManagementLinks(private val links: Map<String, String>) {
+
+    fun linkFor(endpoint: Endpoint): String {
+        return links[endpoint.key] ?: throw ManagementEndpointException(endpoint, "LINK_MISSING")
+    }
+
+    companion object {
+        fun parseManagementResponse(response: JsonNode): ManagementLinks {
```

Comment:
Vi kan bruke spring-hateoas sin Resource klasse for å hjelpe med dette. Den kan brukes inn i RestTemplate, også gir den enkel tilgang til linkene uten å måtte parse json.